src/main/java/dev/binclub/javaception/classfile/ClassFileParser.java

```diff
@@ -1,4 +1,235 @@
 package dev.binclub.javaception.classfile;
 
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import dev.binclub.javaception.classfile.attributes.CodeAttribute;
+import dev.binclub.javaception.classfile.attributes.ExceptionsAttribute;
+import dev.binclub.javaception.classfile.attributes.LineNumberTableAttribute;
+import dev.binclub.javaception.classfile.attributes.LocalVariableTableAttribute;
+import dev.binclub.javaception.classfile.attributes.SourceFileAttribute;
+import dev.binclub.javaception.classfile.attributes.StackMapTableAttribute;
+
 public class ClassFileParser {
+
+	// for testing purposes
+	public static void main(String[] args) throws Throwable {
+		InputStream stream = ClassFileParser.class.getClassLoader().getResourceAsStream("klass/ClassReader.class");
+		ByteArrayOutputStream bos = new ByteArrayOutputStream();
+		int available = 0;
+		while ((available = stream.available()) != 0) {
+			byte[] bytes = new byte[available];
+			stream.read(bytes);
+			bos.write(bytes);
+		}
+		new ClassFileParser(bos.toByteArray());
+
+	}
+
+	// the value of the tag byte acts as the index to get the type
+	public static final ConstantTypes[] types = new ConstantTypes[19];
+
+	int constantPoolCount;
+	int majorVersion;
+	int minorVersion;
+	int access;
+	int interfacesCount;
+	int fieldsCount;
+	int methodsCount;
+	int attributesCount;
+	String className;
+	String superClass;
+	DataInputStream dis;
+	Object[] constantPool;
+	ClassInfoCP[] interfaces;
+	FieldInfo[] fields;
+	MethodInfo[] methods;
+	AttributeInfo[] attributes;
+
+	public ClassFileParser(byte[] classBytes) throws Throwable {
+		if (classBytes.length < 4) {
+			// need atleast 4 bytes get magic number
+			throw new Throwable("Invalid class file");
```

Comment:
This should throw a `ClassFormatError` ([jvms 5.3.5](https://docs.oracle.com/javase/specs/jvms/se16/html/jvms-5.html#jvms-5.3.5)).